daemon/container.go

```diff
@@ -251,6 +251,19 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
 		}
 	}
 
+	p := hostConfig.RestartPolicy
+
+	switch p.Name {
+	case "always", "unless-stopped":
+		if p.MaximumRetryCount != 0 {
+			return nil, fmt.Errorf("maximum restart count not valid with restart policy of '%s'", p.Name)
+		}
+	case "no", "on-failure", "":
+	// do nothing
```

Comment:
I think we need a validation for `on-failure` as well, to check if it's a positive integer. Currently, this succeeds without error-message;

```
docker run --restart on-failure:-20 -d nginx:alpine
```